dht/dht.go

```diff
@@ -618,13 +648,28 @@ func (s *Server) nodeByID(id string) *node {
 	return nil
 }
 
-func (s *Server) handleQuery(source dHTAddr, m Msg) {
+func (s *Server) handleQuery(source DHTAddr, m Msg) {
+	var (
+		newmsg *Msg
+		skip   bool
+	)
 	node := s.getNode(source, m.SenderID())
 	node.lastGotQuery = time.Now()
 	// Don't respond.
 	if s.config.Passive {
 		return
 	}
+	// Call any hooks that may apply
+	if hook, ok := s.hooks[m.Q]; ok {
+		newmsg, skip = hook(&source, node, &m)
+		if newmsg != nil {
+			m = *newmsg
+		}
+	}
+	// Unless told not to by hook, proceed with normal handling.
+	if skip {
```

Comment:
Fixed, changed to propagate. Lead to a stupid bug where I forgot to set propagate to "true" by default, so the Ping test broke. :)